src/averageAges.js

```diff
@@ -55,9 +64,27 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person =>
+    onlyWithSon
+      ? person.sex === 'm' && people.some(mom => mom.name === person.mother)
+      : people.some(mom => mom.name === person.mother));
+
+  const yearsOld = children.map(item => {
+    const mother = people.find(mom => item.mother === mom.name);
+
+    return item.born - mother.born;
+  });
+
+  return yearsOld.reduce((acc, el) => acc + el) / yearsOld.length;
 }
 
+const receiveСenturyOfLife = (obj, century) =>
+  Math.ceil(obj.died / 100) === century && obj.sex === 'm';
```

Comment:
Не розривай колбеки, зменшується читабельність
```suggestion
const receiveСenturyOfLife = (obj, century) => { 
  return Math.ceil(obj.died / 100) === century && obj.sex === 'm';
}
```